[0.12 branch] Backports

maflcko opened the pull request.

Commit:
[rpcwallet] Don't use floating point

Github-Pull: #8317
Rebased-From: 477777f2503e3a56a267556f0fc5091042d93340

ryanofsky committed:
Send tip change notification from invalidateblock

This change is needed to prevent sync_blocks timeouts in the mempool_reorg
test after the sync_blocks update in the upcoming commit
"[qa] Change sync_blocks to pick smarter maxheight".

This change was initially suggested by Suhas Daftuar <sdaftuar@chaincode.com>
in https://github.com/bitcoin/bitcoin/pull/8680#r78209060

Github-Pull: #9196
Rebased-From: 67c6326abd1788e6f411feb4f44b69774e76aae2

TheBlueMatt commented:
If we're not gonna backport segwit, I don't think we want another 12.X release.

On November 23, 2016 4:28:26 AM PST, MarcoFalke <notifications@github.com> wrote:
>
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/bitcoin/bitcoin/pull/9211
>
>-- Commit Summary --
>
>  * [rpcwallet] Don't use floating point
>  * Send tip change notification from invalidateblock
>
>-- File Changes --
>
>    M src/main.cpp (1)
>    M src/wallet/rpcwallet.cpp (6)
>
>-- Patch Links --
>
>https://github.com/bitcoin/bitcoin/pull/9211.patch
>https://github.com/bitcoin/bitcoin/pull/9211.diff
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/9211

MarcoFalke commented:
Right, there won't be a release on the 0.12 branch, so I changed the title and milestone.

Though, we still fix bugs on the 0.12 branch, as the maintenance end is not reached. Not sure if the invalidateblock bugfix qualifies for backport, but lets see what others think.

laanwj committed:
torcontrol: Explicitly request RSA1024 private key

When generating a new service key, explicitly request a RSA1024 one.

The bitcoin P2P protocol has no support for the longer hidden service names
that will come with ed25519 keys, until it does, we depend on the old
hidden service type so make this explicit.

See #9214.

Github-Pull: #9234
Rebased-From: 7d3b627395582ae7c9d54ebdbc68096d7042162b

jonasschnelli committed:
[Wallet] Bugfix: FRT: don't terminate when keypool is empty

Github-Pull: #9295
Rebased-From: c24a4f5981d47d55aa9e4eb40294832a4d38fb80

jonasschnelli committed:
[QA] add fundrawtransaction test on a locked wallet with empty keypool

Github-Pull: #9295
Rebased-From: 1a6eacbf3b7e3d5941fec1154079bbc4678ce861
